### ISSUE TITLE
building the .sif file

### ISSUE THREAD

**Issue by humanpaingeneticslab** [precimed/gsa-mixer#3]
Dear Oleksandr,

I've cloned the rep, then tried to build the .sif file.
Any help appreciated!

Unfortunately, I get an error:
#10 58.88 Building dependency tree...
#10 59.41 Reading state information...
#10 59.51 E: Version '1:2.25.1-1ubuntu3.12' for 'git' was not found
#10 59.52 update-alternatives: error: alternative path /usr/bin/python3 doesn't exist
==>> It's probably chocking @ "#10 59.51"...?

$ ls -al /usr/bin/python3
lrwxrwxrwx 1 root root 9 Oct 27  2020 /usr/bin/python3 -> python3.8
$ uname -a
Linux X-X-X 5.15.0-107-generic #117~20.04.1-Ubuntu

I found this other rep here:
https://github.com/comorment/mixer
From which I cloned, and found the mixer.sif file to work (the one distributed in the rep)!
BUT, that version of mixer doesn't implement some functions; e.g.:
mixer.py: error: invalid choice: 'split_sumstats'

Interesting, so there's two parallel versions of mixer?

Many Thanks!

**Comment by espenhgn** [comment]
Hi @humanpaingeneticslab,
This repository contains the material for GSA-Mixer which was recently published: https://www.nature.com/articles/s41588-024-01771-1

If this is what you want, then you can omit to build the Docker or Singularity image locally, but fetch them from https://github.com/orgs/precimed/packages?repo_name=gsa-mixer

It looks like @ofrei didn't update the documentation for this in this repository. You may look at the similar procedure in the old MiXeR container repository [here](https://github.com/comorment/mixer#update-the-mixersif-container) and update URLs accordingly.

As for the MiXeR development tree, yes, there are indeed multiple parallel projects in different tracks in this GitHub organization; [uni/bivariate MiXeR](https://github.com/precimed/mixer), [trivariate MiXeR](https://github.com/precimed/mix3r), [MiXeR-pred](https://github.com/precimed/mixer-pred)

**Comment by humanpaingeneticslab** [comment]
Thank you @espenhgn for your prompt response!

I followed the instructions there, but alas:

$ docker pull ghcr.io/precimed/gsa-mixer_sif:latest
**unsupported media type application/vnd.oci.empty.v1+json**

$ singularity pull docker://ghcr.io/precimed/gsa-mixer_sif:latest
**FATAL:   While making image from oci registry: error fetching image to cache: while building SIF from layers: packer failed to pack: while unpacking tmpfs: error unpacking rootfs: unpack rootfs: config blob is not correct mediatype application/vnd.oci.image.config.v1+json: application/vnd.oci.empty.v1+json**

At this point any help will be appreciated :-)

**Comment by espenhgn** [comment]
Hi again, @humanpaingeneticslab, 
Unfortunately, we cannot seem to update the GHCR.io autogenerated pulling instructions. For the "_sif" suffix package you need the [Oras](https://oras.land) client, as explained in the other [instructions](https://github.com/comorment/mixer#update-the-mixersif-container) I linked to. 
Either should work (replace `<tag>` with `latest` or some other tag):
```
apptainer pull docker://ghcr.io/comorment/gsa-mixer:<tag>  # or
singularity pull docker://ghcr.io/comorment/gsa-mixer:<tag> # or 
oras pull ghcr.io/comorment/gsa-mixer_sif:<tag>
```
Practically, these are likely similar to the mixer.sif file you obtained, however.

**Comment by ofrei** [comment]
@humanpaingeneticslab sorry for my slow response on this! I now Docker and singulrity builds ([this](https://github.com/precimed/gsa-mixer/commit/faa646d51e70fc070c474fa11336e6ab287ac6c9) was the fix), and updated the [README](https://github.com/precimed/gsa-mixer/blob/main/README.md) file with instructions on how to pull Docker and Singularity versions of the GSA-MiXeR.  Using the latest images  (I've tested both Docker and Singularity) I tested  the instructions  from [Getting Started Example the README.md](https://github.com/precimed/gsa-mixer/blob/main/README.md#getting-started-example), and they run without errors.  I'll possibly need to make some small changes to make real-world example run smoothly. I'll get back on this in #5 . Thanks for submitting the tickets!

 I'm closing  this one as building Docker / singulartiy images should be working fine at this point.
